ui/client/containers/EspApp.js

```diff
@@ -70,52 +72,21 @@ class EspApp extends React.Component {
   environmentAlert(params) {
     if (params && params.content)
       return (
-        <span className="navbar-brand vert-middle ">
-          <span className={`indicator ${  params.cssClass}`}>{params.content}</span>
-        </span>
+          <span className={`indicator ${params.cssClass}`} title={params.content}>{params.content}</span>
       )
   }
 
   render() {
     const AllDialogs = Dialogs.AllDialogs
     return this.props.resolved ? (
       <div id="app-container">
-        <nav id="main-menu" className="navbar navbar-default">
-          <div id="git" className="hide">{JSON.stringify(__GIT__)}</div>
-          <div className="container-fluid">
-            <div className="navbar-header">
-              {this.renderTopLeftButton()}
-              <NavLink id="brand-name" className="navbar-brand" to={EspApp.path}>
-                <span id="app-logo" className="vert-middle">{EspApp.header}</span>
-              </NavLink>
-              {this.environmentAlert(this.props.featuresSettings.environmentAlert)}
-            </div>
-
-            <div className="collapse navbar-collapse">
-              <ul id="menu-items" className="nav navbar-nav navbar-right nav-pills nav-stacked">
-                <li><NavLink to={Processes.path}>{Processes.header}</NavLink></li>
-                <li><NavLink to={SubProcesses.path}>{SubProcesses.header}</NavLink></li>
-                {
-                  !_.isEmpty(this.props.featuresSettings.metrics) ?
-                    <li><NavLink to={Metrics.basePath}>{Metrics.header}</NavLink></li> : null
-                }
-                {
-                  !_.isEmpty(this.props.featuresSettings.search) ?
-                    <li><NavLink to={Search.path}>{Search.header}</NavLink></li> : null
-                }
-                {
-                  this.props.featuresSettings.signals ?
-                    <li><NavLink to={Signals.path}>{Signals.header}</NavLink></li> : null
-                }
-                <li><NavLink to={Archive.path}>{Archive.header}</NavLink></li>
-                {
-                  this.props.loggedUser.globalPermissions.adminTab ?
-                    <li><NavLink to={AdminPage.path}>{AdminPage.header}</NavLink></li> : null
-                }
-              </ul>
-            </div>
-          </div>
-        </nav>
+        <div className="hide">{JSON.stringify(__GIT__)}</div>
+        <MenuBar
+            {...this.props}
+            app={EspApp}
+            leftElement={this.renderTopLeftButton()}
+            rightElement={this.environmentAlert(this.props.featuresSettings.environmentAlert)}
```

Comment:
Maybe extract featuresSettings from props? What if featuresSettings is null? I'm not sure that this field always has value.. :)